LiteDB/Engine/Disk/Serializer/BufferReader.cs

```diff
@@ -251,13 +253,13 @@ private T ReadNumber<T>(Func<byte[], int, T> convert, int size)
             }
             else
             {
-                var buffer = BufferPool.Rent(size);
+                var buffer = bufferPool.Rent(size);
```

Comment:
**Warning!** We need to to zero the loaned buffer! Otherwise the convert will return unexpected results! Rel: [ArrayPool<T>.Rent(Int32) Method](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent?view=net-8.0&devlangs=csharp&f1url=%3FappId%3DDev17IDEF1%26l%3DEN-US%26k%3Dk(System.Buffers.ArrayPool%25601.Rent)%3Bk(DevLang-csharp)%26rd%3Dtrue#remarks)